src/axom/quest/examples/quest_step_file.cpp

```diff
@@ -0,0 +1,2223 @@
+          SLIC_INFO_IF(
+            m_verbose,
+            axom::fmt::format("[Patch {} Wire {} Edge {} Curve {}] Processing "
+                              "edge with curve type: {}",
+                              patchIndex,
+                              wireIndex,
+                              edgeIndex,
+                              curves.size(),
+                              curveTypeStr));
+
+          Standard_Real first, last;
+          Handle(Geom2d_Curve) parametricCurve =
+            BRep_Tool::CurveOnSurface(edge,
+                                      TopoDS::Face(faceExp.Current()),
+                                      first,
+                                      last);
+          Handle(Geom2d_BSplineCurve) bsplineCurve =
+            Geom2dConvert::CurveToBSplineCurve(parametricCurve);
+
+          if(!parametricCurve.IsNull() && !bsplineCurve.IsNull())
+          {
+            const bool originalCurvePeriodic = bsplineCurve->IsPeriodic();
+
+            CurveProcessor curveProcessor(bsplineCurve, m_verbose);
+            curves.emplace_back(curveProcessor.nurbsCurve());
+            patchData.trimmingCurves_originallyPeriodic.push_back(
+              curveProcessor.curveWasOriginallyPeriodic());
+
+            if(isReversed)  // Ensure consistency of curve w.r.t. patch
+            {
+              curves.back().reverseOrientation();
+            }
+            SLIC_ASSERT(curves.back().isValidNURBS());
+            SLIC_ASSERT(curves.back().getDegree() == bsplineCurve->Degree());
+
+            SLIC_INFO_IF(
+              m_verbose,
+              axom::fmt::format(
+                "[Patch {} Wire {} Edge {} Curve {}] Added curve: {}",
+                patchIndex,
+                wireIndex,
+                edgeIndex,
+                curves.size(),
+                curves.back()));
+
+            // Check to ensure that curve did not change geometrically after making non-periodic
+            if(originalCurvePeriodic)
+            {
+              Handle(Geom2d_BSplineCurve) origCurve =
+                Geom2dConvert::CurveToBSplineCurve(parametricCurve);
+              const bool withinThreshold =
+                curveProcessor.compareToCurve(origCurve, 25);
+              SLIC_WARNING_IF(
+                !withinThreshold,
+                axom::fmt::format(
+                  "[Patch {} Wire {} Edge {} Curve {}] Trimming curve was not "
+                  "within threshold after clamping.",
+                  patchIndex,
+                  wireIndex,
+                  edgeIndex,
+                  curves.size(),
+                  curves.back()));
+            }
+          }
+        }
+      }
+    }
+  }
+
+  const PatchDataMap& getPatchDataMap() const { return m_patchData; }
+
+  std::string getFileUnits() const { return m_fileUnits; }
+
+private:
+  /// Returns the canoninical representation of a unit string (e.g. "centimeter" -> "cm")
+  std::string getCanonicalUnit(const std::string& unit) const
```

Comment:
Consider moving units code to axom::core unless it is specific to STEP.

Reply:
That's a good idea! Do you think it's general enough? I'd want to make this more robust for a general utility. I'll create an issue for this and move it and test it in a follow-up PR.

Reply:
See: https://github.com/LLNL/axom/issues/1689